ksqldb-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java

```diff
@@ -370,10 +371,13 @@ private Config getConfig() {
     return KafkaClusterUtil.getConfig(adminClient.get());
   }
 
+  @SuppressWarnings("checkstyle:BooleanExpressionComplexity")
   private static boolean isInternalTopic(final String topicName, final String applicationId) {
     return topicName.startsWith(applicationId + "-")
         && (topicName.endsWith(KsqlConstants.STREAMS_CHANGELOG_TOPIC_SUFFIX)
-        || topicName.endsWith(KsqlConstants.STREAMS_REPARTITION_TOPIC_SUFFIX));
+        || topicName.endsWith(KsqlConstants.STREAMS_REPARTITION_TOPIC_SUFFIX))
+        || topicName.matches(KsqlConstants.STREAMS_JOIN_REGISTRATION_TOPIC_PATTERN)
+        || topicName.matches(KsqlConstants.STREAMS_JOIN_RESPONSE_TOPIC_PATTERN);
```

Comment:
This expression does not seem completely correct. Currently it is `A && (B || C) || D || E`. Shouldn't it be `A && (B || C || D || E)`? Maybe that is also the reason you get a checkstyle error that you need to suppress.
If I am right, then we are missing some tests that check the negative case.

Reply:
Good catch! I fixed it and added checks for the negative cases in 9e2652e.